blogpost/blogpost.md

```diff
@@ -0,0 +1,845 @@
+      try {
+        return objectMapper.readValue(record.value(), Lightning.class);
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    }).collect(Collectors.toList());
+  }
+
+  private Consumer<Long, String> createConsumer() {
+    Properties props = new Properties();
+    props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, kafkaContainer.getBootstrapServers());
+    props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, LongDeserializer.class.getName());
+    props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
+    props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+    props.put(ConsumerConfig.GROUP_ID_CONFIG, "lightning_group");
+    KafkaConsumer<Long, String> consumer = new KafkaConsumer<>(props);
+    consumer.subscribe(Collections.singletonList(KAFKA_TOPIC));
+    return consumer;
+  }
+
+  public static Environment mockEnvironment(String bootstrapServers) {
+    MockEnvironment environment = new MockEnvironment();
+    environment.setProperty("beam.window_size", "5");
+    environment.setProperty("kafka.bootstrap.servers", bootstrapServers);
+    environment.setProperty("kafka.topic", KAFKA_TOPIC);
+    environment.setProperty("kafka.auto.offset.reset", "earliest");
+    environment.setProperty("kafka.limit", "2");
+    return environment;
+  }
+
+  private PipelineOptions pipelineOptions() {
+    PipelineOptions options = PipelineOptionsFactory.create();
+    options.setRunner(FlinkRunner.class);
+    return options;
+  }
+}
+```
+
+This test doesn't need Spring context so I can just mock the environment and create the Bean class without autowiring anything.
+Then the emulator sends some limited number of records and I read them using the KafkaConsumer. There is really not much to explain more.
+
+## How to use the app
+### Run it
+Download the sources and run the following commands from the app root directory:
+```
+docker-compose up
+```
+This will set up the Kafka and MongoDB instances
+```
+./mvnw spring-boot:run
+```
+This will build and run the application.
+
+You can also open the project in IntelliJ and run it via `LightningApplication` main method.
+
+### Use it
+
+The app exposes two endpoints: 
+```
+http://localhost:8080/lightning
+``` 
+with one possible parameter - timestamp in milliseconds. It gives us all lightning json data with timestamp later than
+the one from the param. Example:
+```
+http://localhost:8080/lightning?timestamp=1600760252089
+```
+To get strikes data use this url:
+```
+http://localhost:8080/strikes
+```
+
+## Summary
+
+That's all! I presented a Spring Boot application that exposes two endpoints providing real-time data supported with a Apache Beam streaming pipeline running Apache Flink under the hood reading from Apache Kafka to MongoDB.
+
+You can find the whole source code at [GitHub](https://github.com/piotr-szuberski/lightning).
```

Comment:
Słyszałem coś o istnieniu dedykowanemu blog postom miejsca na Githuba/Gitlabie. Jeśli nie chcesz być zmuszony do trzymania tego repo u siebie, to podpytaj o szczegóły Ulę albo Kasię

Reply:
Jak coś to bardzo chętnie potrzymam u siebie - zawsze można podmienić później jakby się okazało że lepiej. Widziałem że Michał swój kod wrzucił do siebie :D